modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp

```diff
@@ -151,113 +151,130 @@ if (portletTitleBasedNavigation) {
+											</liferay-portlet:resourceURL>
 
-												itemFailed = true;
+											if (commonFileMetadataContainer.io) {
+												commonFileMetadataContainer.io.start();
+											}
+											else {
+												commonFileMetadataContainer.load(
+													'<%= uploadMultipleFileEntries %>'
+												);
 											}
 
-											li.addClass(cssClass);
-											li.append(childHTML);
-										}
+											Liferay.fire('filesSaved');
 
-										<liferay-portlet:resourceURL copyCurrentRenderParameters="<%= false %>" id="/document_library/upload_multiple_file_entries" var="uploadMultipleFileEntries">
-											<portlet:param name="repositoryId" value="<%= String.valueOf(repositoryId) %>" />
-											<portlet:param name="folderId" value="<%= String.valueOf(folderId) %>" />
-										</liferay-portlet:resourceURL>
+											commonFileMetadataContainer.unplug(A.LoadingMask);
 
-										if (commonFileMetadataContainer.io) {
-											commonFileMetadataContainer.io.start();
-										}
-										else {
-											commonFileMetadataContainer.load(
-												'<%= uploadMultipleFileEntries %>'
+											if (!itemFailed) {
+												location.href = '<%= HtmlUtil.escapeJS(redirect) %>';
+											}
+										})
+										.catch((error) => {
+											var selectedItems = A.all(
+												'#<portlet:namespace />fileUpload li.selected'
 											);
-										}
-
-										Liferay.fire('filesSaved');
 
-										commonFileMetadataContainer.unplug(A.LoadingMask);
+											selectedItems
+												.removeClass('selectable')
+												.removeClass('selected')
+												.addClass('upload-error');
 
-										if (!itemFailed) {
-											location.href = '<%= HtmlUtil.escapeJS(redirect) %>';
-										}
-									})
-									.catch((error) => {
-										var selectedItems = A.all(
-											'#<portlet:namespace />fileUpload li.selected'
-										);
+											selectedItems.append(
+												'<span class="card-bottom error-message"><%= UnicodeLanguageUtil.get(request, "an-unexpected-error-occurred-while-deleting-the-file") %></span>'
+											);
 
-										selectedItems
-											.removeClass('selectable')
-											.removeClass('selected')
-											.addClass('upload-error');
+											selectedItems.all('input').remove(true);
 
-										selectedItems.append(
-											'<span class="card-bottom error-message"><%= UnicodeLanguageUtil.get(request, "an-unexpected-error-occurred-while-deleting-the-file") %></span>'
-										);
+											commonFileMetadataContainer.loadingmask.hide();
+										});
+								};
 
-										selectedItems.all('input').remove(true);
+								var isDataEngineControlled = Boolean(
+									document.querySelector('[data-ddm-fieldset]')
+								);
 
-										commonFileMetadataContainer.loadingmask.hide();
+								if (isDataEngineControlled) {
+									Liferay.on('ddmFormValid', (event) => {
```

Comment:
Shouldn't we detach this event at some point?

Reply:
Thanks, you are right :)

Reply:
Detach events here dea61e9775094b5a00ddd701dfbe60718943d685